### ISSUE TITLE
Isn't multiclass and multilabel switched?

### ISSUE THREAD

**Issue by marcoramosw** [qubvel/segmentation_models.pytorch#369]
I am looking at the Dice loss function and I am having a hard time figuring out why you do one hot encoding in multiclass and not on multilabel, it seems to me that the code in multiclass is for multilabel and multilabel for multiclass

`

            if self.mode == MULTICLASS_MODE:
            y_true = y_true.view(bs, -1)
            y_pred = y_pred.view(bs, num_classes, -1)

            y_true = F.one_hot(y_true, num_classes)  # N,H*W -> N,H*W, C
            y_true = y_true.permute(0, 2, 1)  # H, C, H*W

        if self.mode == MULTILABEL_MODE:
            y_true = y_true.view(bs, num_classes, -1)
            y_pred = y_pred.view(bs, num_classes, -1)`

**Comment by SippieCup** [comment]
Diceloss function in this repo is broken for multiclass. Just import and use the updated dice loss from [pytorch-toolbelt](https://github.com/BloodAxe/pytorch-toolbelt)
